cmd/kubeadm/app/apis/kubeadm/v1alpha2/conversion.go

```diff
@@ -32,103 +32,144 @@ func Convert_v1alpha2_InitConfiguration_To_kubeadm_InitConfiguration(in *InitCon
 	if err := autoConvert_v1alpha2_InitConfiguration_To_kubeadm_InitConfiguration(in, out, s); err != nil {
 		return err
 	}
+	if err := split_v1alpha2_InitConfiguration_into_kubeadm_ClusterConfiguration(in, &out.ClusterConfiguration, s); err != nil {
+		return err
+	}
+	return nil
+}
 
+func split_v1alpha2_InitConfiguration_into_kubeadm_ClusterConfiguration(in *InitConfiguration, out *kubeadm.ClusterConfiguration, s conversion.Scope) error {
+	if err := split_v1alpha2_InitConfiguration_into_kubeadm_ComponentConfigs(in, &out.ComponentConfigs, s); err != nil {
```

Comment:
Do we have UT that covers the conversion tests?

Reply:
Also, I'm not certain why this code is under v1alpha2 ClusterConfiguration is only v1alpha3.

Reply:
> Do we have UT that covers the conversion tests?

For tests I followed Luxas approach already used for API changes during last cycle.

There is a robust and estensive integration test under `cmd/kubeadm/app/util/config` (but no UT for each function strinctly speaking); with the integration test we are doing the following roundtrips back and forth different API formats
- v1alpha2 -> internal
- v1alpha3 -> internal
- v1alpha2 -> internal -> v1alpha3
- v1alpha3 -> internal -> v1alpha3
- v1alpha2 -> default -> validate -> internal -> v1alpha3

And then we check that no data are lost, so I'm pretty confident that everything is covered.